ext/phar/util.c

```diff
@@ -1482,12 +1490,22 @@ int phar_verify_signature(php_stream *fp, size_t end_of_phar, uint32_t sig_type,
 	php_stream_rewind(fp);
 
 	switch (sig_type) {
+		case PHAR_SIG_OPENSSL_512:
+		case PHAR_SIG_OPENSSL_256:
 		case PHAR_SIG_OPENSSL: {
 #ifdef PHAR_HAVE_OPENSSL
 			BIO *in;
 			EVP_PKEY *key;
-			EVP_MD *mdtype = (EVP_MD *) EVP_sha1();
+			EVP_MD *mdtype;
```

Comment:
Change this to `const EVP_MD *` and drop casts below?

Reply:
done